keps/sig-storage/5040-remove-gitrepo-driver/README.md

```diff
@@ -0,0 +1,1084 @@
+However, if complete unit test coverage is not possible, explain the reason of it
+together with explanation why this is acceptable.
+-->
+
+<!--
+Additionally, for Alpha try to enumerate the core package you will be touching
+to implement this enhancement and provide the current unit coverage for those
+in the form of:
+- <package>: <date> - <current test coverage>
+The data can be easily read from:
+https://testgrid.k8s.io/sig-testing-canaries#ci-kubernetes-coverage-unit
+
+This can inform certain test coverage improvements that we want to do before
+extending the production code to implement this enhancement.
+-->
+
+<!-- - `<package>`: `<date>` - `<test coverage>` -->
+
+We will add the following unit tests to the [git_repo](https://github.com/kubernetes/kubernetes/tree/master/pkg/volume/git_repo/git_repo_test.go) 
+pacakage:
+
+- When the feature-gate GitRepoVolumeDriver is false kubelet returns an error
+  for pods that have gitRepo volumes in them
+- When the feature-gate GitRepoVolumeDriver is false kubelet does not return an
+  error for pods that have gitRepo volumes in them
+- When the feature-gate GitRepoVolumeDriver is true kubelet does not returns an
+  error for pods that have gitRepo volumes in them
+
+##### Integration tests
+
+<!--
+Integration tests are contained in k8s.io/kubernetes/test/integration.
+Integration tests allow control of the configuration parameters used to start the binaries under test.
+This is different from e2e tests which do not allow configuration of parameters.
+Doing this allows testing non-default options and multiple different and potentially conflicting command line options.
+-->
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+-->
+
+- <test>: <link to test coverage>
+
+##### e2e tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+
+We expect no non-infra related flakes in the last month as a GA graduation criteria.
+-->
+
+We will add the following e2e tests to the [e2e_node](https://github.com/kubernetes/kubernetes/tree/master/test/e2e_node) 
+pacakage:
+
+- When the feature-gate GitRepoVolumeDriver is false kubelet returns an error
+  for pods that have gitRepo volumes in them
+- When the feature-gate GitRepoVolumeDriver is false kubelet does not return an
+  error for pods that have gitRepo volumes in them
+- When the feature-gate GitRepoVolumeDriver is true kubelet does not returns an
+  error for pods that have gitRepo volumes in them
+
+### Graduation Criteria
+
+<!--
```

Comment:
Can you do a pass and remove the comment blocks from sections which you believe are done?  Makes reading the MD source easier :)

Reply:
Done.